app/static/js/pico_recipe.js

```diff
@@ -122,13 +131,17 @@ var recipe_table = {
         {
             formatter: plusIcon, width: 49, hozAlign: "center",
             cellClick: function (e, cell) {
-                cell.getTable().addRow({}, false, cell.getRow());
+                if (rowIsEditable(cell))
+                    cell.getTable().addRow(Object.assign({}, cell.getRow().getData()), false, cell.getRow()).then(function(row) {
+                        row.update({name: "New Step"});
+                    });
             }
         },
         {
             formatter: minusIcon, width: 49, hozAlign: "center",
             cellClick: function (e, cell) {
-                cell.getRow().delete();
+                if (rowIsEditable(cell))
+                    cell.getRow().delete();

```

Comment:
like these improvements for the pico recipes thanks for essentially "locking" these rows from user editing.

Reply:
I tried for a while to make the first 3 rows for the pico not movable, but that was the best i could do

Reply:
I couldn't see a way to easily do it either. Maybe a validation of the table could be added to the callback for a move to make sure that nothing was moved into or out of the first 3 rows, but I also couldn't see an easy way to undo the move if it wasn't valid.